### ISSUE TITLE
How to deal with global entities ( singleton, constants  etc.. ) in R

### ISSUE THREAD

**Issue by msevestre** [Open-Systems-Pharmacology/2R#64]

**Comment by msevestre** [comment]
I like the idea of stateful function functions
https://adv-r.hadley.nz/function-factories.html#stateful-funs

also check the rest of the article related to memory allocation and collection

**Comment by PavelBal** [comment]
I think there is no way to completely avoid global variables, unless we want to force users to use objects (and wo don't). The key is to make those variables not visible to the user, e.g., by saving them in a separate environment.
https://www.r-bloggers.com/global-variables-in-r-packages/

http://adv-r.had.co.nz/Environments.html

I will read more about how to organize it.

**Comment by msevestre** [comment]
>Unless we want to force users to use objects  

Can you elaborate on what you mean here?

**Comment by PavelBal** [comment]
I think this can be closed?

**Comment by msevestre** [comment]
No. I am still not certain as to what is the best way to store global constants but private to the package.
